src/apollo/zimbra-in-memory-cache.ts

```diff
@@ -22,10 +22,13 @@ const dataIdFromObject = (object: any) => {
 		case 'Folder':
 			if (object.local && object.local !== null) {
 				return `${defaultDataIdFromObject(object)}:local`;
-			}
-			if (object.id === '1') {
+			} else if (object.id === '1') {
 				// Cache the root folder based on both ID and UUID from server
 				return `${object.__typename}:${object.id}:${object.uuid}`;
+			} else if (object?.id?.includes(':')) {
+				// Cache the root folder based on both ID and UUID from server
+				let i = `${object.id}:${object.absFolderPath}`;
+				return i;
```

Comment:
- [ ] combine both statements
- [ ] also fix comment
- [ ] absFolderPath can contain special characters, which can break caching mechanism, how would you handle that?

Reply:
First two comments code change is done.
For third point I have checked with `${{(PAE??SI**~1!@#%<><>))+^==^` seems like working fine. And we cannot use colon, slash and double-quote in name